opennms-install/src/main/java/org/opennms/install/Installer.java

```diff
@@ -572,6 +577,56 @@ public void verifyFilesAndDirectories() throws FileNotFoundException {
             verifyFileExists(true, m_install_servletdir, "OpenNMS servlet directory",
                              "install.servlet.dir property");
         }
+
+        final var validator = new FilesystemPermissionValidator();
+        final var user = getRunas();
+
+        final Path opennmsHome = Paths.get(m_opennms_home);
+        try {
+            validator.validate(user, opennmsHome);
+        } catch (final FilesystemPermissionException e) {
+            LOG.error("OpenNMS is configured to run as '{}' but '{}' is not owned by that account.", user, e.path);
+            LOG.error("To fix permissions, run '{}/bin/fix-permissions' as root", m_opennms_home);
+            System.out.println();
+
+            if (m_validate_ownership) {
+                System.exit(1);
+            }
+        }
+    }
+
+    private String getRunas() throws IOException {
+        if (RUNAS_USER == null) {
+            final var opennmsConf = readOpennmsConf();
+            // use RUNAS from opennms.conf if found, fall back to -Dopennms.runas, fall back to "opennms" if nothing is found
+            RUNAS_USER = opennmsConf.getProperty("RUNAS", System.getProperty("opennms.runas", "opennms"));
+        }
+
+        return RUNAS_USER;
+    }
+
+    private Properties readOpennmsConf() throws IOException {
```

Comment:
Good candidate for a unit test? Feels like there are a lot of edge cases in here.

Reply:
Good point, I'll add one.